sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cra-integration/cra-process-integration.scheduler.ts

```diff
@@ -0,0 +1,47 @@
+import { InjectQueue, Process, Processor } from "@nestjs/bull";
+import { CRAPersonalVerificationService } from "@sims/integrations/cra-integration/cra-personal-verification.service";
+import { QueueService } from "@sims/services/queue";
+import { QueueNames } from "@sims/utilities";
+import { InjectLogger, LoggerService } from "@sims/utilities/logger";
+import { Job, Queue } from "bull";
+import { BaseScheduler } from "../base-scheduler";
+import { CRAValidationResultQueueOutDTO } from "./models/cra-validation-result.dto";
+
+@Processor(QueueNames.CRAProcessIntegration)
+export class CRAProcessIntegrationScheduler extends BaseScheduler<void> {
+  constructor(
+    @InjectQueue(QueueNames.CRAProcessIntegration)
+    protected readonly schedulerQueue: Queue<void>,
+    protected readonly queueService: QueueService,
+    private readonly cra: CRAPersonalVerificationService,
+  ) {
+    super(schedulerQueue, queueService);
+  }
+
+  /**
+   * Identifies all the student applications that have a pending
+   * income verification and generate the request file to be
+   * processed by CRA.
+   * @params job job details.
+   * @returns Processing result log.
+   */
+  @Process()
+  async processIncomeVerification(
+    job: Job<void>,
+  ): Promise<CRAValidationResultQueueOutDTO> {
+    this.logger.log(
+      `Processing CRA integration job ${job.id} of type ${job.name}.`,
+    );
+    this.logger.log("Executing income validation...");
+    const uploadResult = await this.cra.createIncomeVerificationRequest();
+    this.logger.log("Income validation executed.");
+    await this.cleanSchedulerQueueHistory();
```

Comment:
It will be good to add a TODO to queues which are not schedulers to add this clean up history. For now we have only one queue which is start-application-assessment

Reply:
add a todo to where? these are schedulers, right? oh did u mean to add to ` start-application-assessment` ??